src/App.tsx

```diff
@@ -1,44 +1,77 @@
-import React from 'react';
+import React, { useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
 import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
-import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
-import { Loader } from './components/Loader';
+import { useSelectedUser } from './components/Contexts/UserContext';
+import { PostDetails } from './components/PostDetails';
+import { Post } from './types/Post';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+
+  const [isNewCommentActive, setIsNewCommentActive] = useState(false);
+  const [isDropDownActive, setIsDropDownActive] = useState(false);
+
+  const [postsLoadingError, setPostsLoadingError] = useState(false);
+
+  const { selectedUser, setSelectedUser } = useSelectedUser();
+
+  const changeSelectedUser = (newUser: User) => {
+    setSelectedPost(null);
+    setSelectedUser(null);
+    setSelectedUser(newUser);
+  };
+
   return (
-    <main className="section">
+    <main
+      className="section"
+      role="presentation"
+      onClick={() => {
+        if (isDropDownActive) {
+          setIsDropDownActive(false);
+        }
+      }}
+    >
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  isDropDownActive={isDropDownActive}
+                  onCloseDropDown={setIsDropDownActive}
+                  onChangeUser={changeSelectedUser}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
-
-                <Loader />
-
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
-
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
-
-                <PostsList />
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
+                {postsLoadingError && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                )}
+                {(selectedUser && !postsLoadingError) && (
```

Comment:
For better readability, conditions where there are more than two values should be assigned to separate variables.